src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_1_TypicalWorkflow
+    {
+        [Workflow(runMethod: nameof(SayHelloAsync))]
+        public class SayHelloWorkflow
+        {
+            private const string AddresseeNameDefault = "Boss";
+
+            private string _addresseeName = null;
+
+            public async Task SayHelloAsync(WorkflowContext workflowCtx)
```

Comment:
As @bergundy mentioned before, since we control the runtime, I think we should make the context implicit. We sure would in Go if we could.

Reply:
Adding an implicit context value can always be done, but once there, it cannot be taken away easily. Providing such magic comes with caveats: testability is one example. I would prefer to start with explicit and add implicit in a subsequent version, if there is demand for it.

Reply:
Strongly disagree, let's put implicit context in the proposal or if we think it's impossible to support at least document why.